client/src/app/hooks/useSelectionState/useSelectionState.ts

```diff
@@ -4,99 +4,89 @@ export interface ISelectionStateArgs<T> {
-  isItemSelectable = () => true,
-  externalState,
 }: ISelectionStateArgs<T>): ISelectionState<T> => {
-  const internalState = React.useState<T[]>(initialSelected);
-  const [selectedItems, setSelectedItems] = externalState || internalState;
-
-  const selectableItems = React.useMemo(
-    () => items.filter(isItemSelectable),
-    [items, isItemSelectable]
-  );
+  const [selectedItems, setSelectedItems] =
+    React.useState<T[]>(initialSelected);
 
   const isItemSelected = React.useCallback(
     (item: T) => selectedItems.some((i) => isEqual(item, i)),
     [isEqual, selectedItems]
   );
 
-  // If isItemSelectable changes and a selected item is no longer selectable, deselect it
-  React.useEffect(() => {
-    if (!selectedItems.every(isItemSelectable)) {
-      setSelectedItems(selectedItems.filter(isItemSelectable));
-    }
-  }, [isItemSelectable, selectedItems, setSelectedItems]);
-
-  const toggleItemSelected = React.useCallback(
-    (item: T, isSelecting = !isItemSelected(item)) => {
-      if (isSelecting && isItemSelectable(item)) {
+  const selectItem = React.useCallback(
+    (item: T, isSelecting) => {
+      if (isSelecting) {
         setSelectedItems([...selectedItems, item]);
       } else {
         setSelectedItems(selectedItems.filter((i) => !isEqual(i, item)));
       }
     },
-    [isEqual, isItemSelectable, isItemSelected, selectedItems, setSelectedItems]
+    [isEqual, selectedItems, setSelectedItems]
   );
 
-  const selectMultiple = React.useCallback(
+  const selectItems = React.useCallback(
     (itemsSubset: T[], isSelecting: boolean) => {
       const otherSelectedItems = selectedItems.filter(
         (selected) => !itemsSubset.some((item) => isEqual(selected, item))
       );
-      const itemsToSelect = itemsSubset.filter(isItemSelectable);
       if (isSelecting) {
-        setSelectedItems([...otherSelectedItems, ...itemsToSelect]);
+        setSelectedItems([...otherSelectedItems, ...itemsSubset]);
       } else {
         setSelectedItems(otherSelectedItems);
       }
     },
-    [isEqual, isItemSelectable, selectedItems, setSelectedItems]
+    [isEqual, selectedItems, setSelectedItems]
+  );
+
+  const selectOnly = React.useCallback(
+    (toSelect: T[]) => {
+      const filtered = items.filter((item) =>
+        toSelect.find((sel) => isEqual(item, sel))
+      );
+      setSelectedItems(filtered);
+    },
+    [isEqual, items]
   );
 
   const selectAll = React.useCallback(
-    (isSelecting = true) =>
-      setSelectedItems(isSelecting ? selectableItems : []),
-    [selectableItems, setSelectedItems]
+    (isSelecting) => setSelectedItems(isSelecting ? items : []),
+    [items, setSelectedItems]
   );
-  const areAllSelected = selectedItems.length === selectableItems.length;
+  const areAllSelected = selectedItems.length === items.length;
 
-  // Preserve original order of items
-  const selectedItemsInOrder = React.useMemo(() => {
+  const selectedItemsInItemsOrder = React.useMemo(() => {
```

Comment:
WDYT about combining this step with setting the selection state? i.e. 
This would ensure that we have one consistent selection state.
```
setSelectedInItemsOrder= (toBeSelected) => setSelectedItems(
    items.filter(it => toBeSelected.some(selected => ...)
)
```